pkg/storegateway/gateway_blocks_http.go

```diff
@@ -74,6 +97,73 @@ func (s *StoreGateway) BlocksHandler(w http.ResponseWriter, req *http.Request) {
 	showDeleted := req.Form.Get("show_deleted") == "on"
 	showSources := req.Form.Get("show_sources") == "on"
 	showParents := req.Form.Get("show_parents") == "on"
+	actionType := req.Form.Get("action_type")
+	if actionType == "" {
+		actionType = string(ActionTypeNone) // Set the default value to "none"
+	}
+
+	if !isValidActionType(ActionType(actionType)) {
+		// Handle the case where the parsed value is not a valid action type
+		util.WriteTextResponse(w, fmt.Sprintf("Invalid Action Type: %s\n", actionType))
+		return
+	}
+	action := ActionType(actionType)
+	// When blockUids is set, and dropdown action is "no-compact" or "delete-no-compact",
+	// we will perform the action on the selected blocks
+	if (action == ActionTypeNoCompact || action == ActionTypeDeleteNoCompact) && req.Form.Get("block_uids") != "" {
+		var uids []string
+		// URL-decode the string
+		decodedString, err := url.QueryUnescape(req.Form.Get("block_uids"))
+		if err != nil {
+			util.WriteTextResponse(w, fmt.Sprintf("Error decoding URL: %s", err.Error()))
+			return
+		}
+
+		decodedBytes, err := base64.StdEncoding.DecodeString(decodedString)
+		if err != nil {
+			util.WriteTextResponse(w, fmt.Sprintf("Can't decode base64 of selected blocks' uid: %s", err))
+			return
+		}
+		if err := json.Unmarshal(decodedBytes, &uids); err != nil {
+			util.WriteTextResponse(w, fmt.Sprintf("Can't parse block_uids: %s", err))
+			return
+		}
+		for _, uid := range uids {
+			ulid, err := ulid.Parse(uid)
+			if err != nil {
+				util.WriteTextResponse(w, fmt.Sprintf("Can't parse ULID %s: %s", uid, err))
+				return
+			}
+			bkt := block.BucketWithGlobalMarkers(bucket.NewUserBucketClient(tenantID, s.stores.bucket, nil))
+			if action == ActionTypeNoCompact {
+				err = block.MarkForNoCompact(
```

Comment:
let's add some logging trace that this marker was uploaded by some user. Perhaps adding a special logging line in this function.

in addition to the tenant ID if we can get their IP that would be great, but I'm not sure how easy it will be. Don't want to block the PR because of the IP

Reply:
added.